### PR TITLE
rpl: make PIOs in DIOs configurable and send them out always or never

### PR DESCRIPTION
**Pull request by cgundogan**
Currently, the RPL implementation sends out PIOs within DIOs only for the first DIO after joining/creating a DODAG.

This PR changes the behaviour in the way that PIOs are always sent out within DIOs. I also added a way to configure the transmission of PIOs (always or never) either via the shell command `rpl set pio <on/off> <instance_id>` or by excluding the code with `CFLAGS += -DGNRC_RPL_WITHOUT_PIO`.

It is not necessary to have PIOs in DIOs when a border router is providing PIOs in RAs.

### CODE REVIEW

**Comment by OlegHahm** [comment]
:+1: - will test.

File: sys/include/net/gnrc/rpl.h [hunk 1]
```diff
@@ -12,6 +12,26 @@
...
  * @brief       RPL implementation for GNRC
+ *
+ * Configuration
+ * =============
+ *
+ * USEMODULE
+ * ---------
+ *
+ * - RPL (Storing Mode)
+ *   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.mk}
```
Comment (OlegHahm) [inline sys/include/net/gnrc/rpl.h]: What's the purpose of this syntax?

Comment (cgundogan) [inline sys/include/net/gnrc/rpl.h]: code blocks, see here: https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/gnrc.h#L192

Comment (cgundogan) [inline sys/include/net/gnrc/rpl.h]: usage of **```** was broken for some specific versions of doxygen

Comment (OlegHahm) [inline sys/include/net/gnrc/rpl.h]: thx

**Comment by OlegHahm** [comment]
Btw. testing will have to wait until next week. I don't have a sniffer here.

File: sys/shell/commands/sc_gnrc_rpl.c [hunk 2]
```diff
@@ -283,18 +301,35 @@ int _gnrc_rpl(int argc, char **argv)
...
+    else if (strcmp(argv[1], "set") == 0) {
+        if (argc > 2) {
+#ifndef GNRC_RPL_WITHOUT_PIO
+            if (strcmp(argv[2], "pio") == 0) {
+                if ((argc == 5) && (strcmp(argv[3], "on") == 0)) {
+                    return _gnrc_rpl_set_pio(argv[4], true);
+                }
+                else if ((argc == 5) && (strcmp(argv[3], "off") == 0)) {
+                    return _gnrc_rpl_set_pio(argv[4], false);
+                }
```
Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_rpl.c]: else print current state?

**Comment by OlegHahm** [comment]
Tested. And worked.

File: sys/shell/commands/sc_gnrc_rpl.c (hunk 2 above)
Comment (cgundogan) [inline sys/shell/commands/sc_gnrc_rpl.c]: In the else case it currently slides down to the list of commands (`help`). Is this also okay?

Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_rpl.c]: Something that would print if PIO sending is enabled or not would be preferrable, but I can live without.

Comment (cgundogan) [inline sys/shell/commands/sc_gnrc_rpl.c]: sounds interesting. I can extend the output of `rpl` in a follow-up to show the PIO status for all dodags/instances

**Comment by OlegHahm** [comment]
Let's go. Waiting for the follow-up.

**Pull request merged**